### PR TITLE
fairmq: Add v1.7.0

### PR DESCRIPTION
**Pull request by dennisklein**

### CODE REVIEW

File: var/spack/repos/builtin/packages/fairmq/package.py [hunk 1]
```diff
@@ -33,6 +40,7 @@ class Fairmq(CMakePackage):
     # https://github.com/spack/spack/issues/19972
     # https://github.com/spack/spack/issues/14344
 
+    variant("autobind", default=True, description="Override the channel autoBind default")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/fairmq/package.py]:
```suggestion
    variant("autobind", default=True, when="@1.7:", description="Override the channel autoBind default")
```

File: var/spack/repos/builtin/packages/fairmq/package.py [hunk 2]
```diff
@@ -59,11 +67,15 @@ class Fairmq(CMakePackage):
     depends_on("fairlogger@1.6: +pretty")
     depends_on("libzmq@4.1.4:")
 
+    conflicts("@:1.6", when="~autobind")
+
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/fairmq/package.py]:
```suggestion
```

**Review by alalazo — CHANGES_REQUESTED** [review]
We can use a conditional variant in this case

File: var/spack/repos/builtin/packages/fairmq/package.py (hunk 1 above)
Comment (dennisklein) [reply var/spack/repos/builtin/packages/fairmq/package.py]: cool, thx for the hint! done

**Review by alalazo — APPROVED** [review]

**Review by ChristianTackeGSI — APPROVED** [review]

**Pull request merged**